reference-implementation/include/emitc/tosa.h

```diff
@@ -114,6 +114,46 @@ inline Src reluN(Src operand, typename Src::value_type max_value) {
   return emitc::clamp(min, operand, max);
 }
 
+// RescaleOp
+template <typename Dest, size_t Dim, typename Src>
+inline Dest rescale(Src x, typename get_element_type<Src>::type in_zp,
+                    typename get_element_type<Dest>::type out_zp,
+                    Tensor1D<int64_t, Dim> mult, Tensor1D<int64_t, Dim> shift,
+                    bool scale32, bool double_round, bool per_channel) {
+  using ET_Dest = typename get_element_type<Dest>::type;
+  using Dest_I32 = typename replace_element_type<int32_t, Dest>::type;
+
+  double_round &= scale32;
```

Comment:
Could you replace this with a runtime assertion?

The [docs](https://www.mlplatform.org/tosa/tosa_spec.html#_rescale) state that `!scale32 && double_round` should result in an error.

```suggestion
  if(!scale32 && double_round) {
    assert(false && "Invalid combination of `scale32` and `double_round` arguments.");  
  }
```

That makes it easier to relate the code to the docs.

Reply:
Yes, but why not put the error condition inside the assertion? Should I check for the other error cases from the docs, too?

Reply:
We decided that only checking this case will be the most consistent.